dart/dynamics/MetaSkeleton.hpp

```diff
@@ -71,6 +71,11 @@ class MetaSkeleton : public common::Subject
   /// Default destructor
   virtual ~MetaSkeleton() = default;
 
+  /// Returns mutex.
+  virtual std::unique_ptr<common::Mutex> getCustomMutex() const = 0;
```

Comment:
It feels a little bit strange to me to return a `std::unique_ptr` to a "Mutex", since that could create the impression that no one else has access to this particular "Mutex", and therefore locking this "Mutex" would be frivolous. Of course the reality is that this is a `std::unique_ptr` to a wrapper that contains references to mutex(es), so it's perfectly sensible to receive a `std::unique_ptr` to it.

I wonder if we should consider naming the class `MutexWrapper` or `MutexReference`? I'm totally fine with just keeping it as `Mutex`; this was just a thought that hit me while looking through the code, so I wanted to express it in case anyone else was struck with a similar impression.

Reply:
This sounds good to me. I changed the name to `MutexReference` in the similar fashion of `ReferentialSkeleton`, but am open to `MutexWrapper` or better name if any.